libmproxy/proxy/root_context.py

```diff
@@ -47,6 +74,7 @@ def next_layer(self, top_layer):
         layer = self._next_layer(top_layer)
         return self.channel.ask("next_layer", layer)
 
+    @tls_sni_check_ignore
```

Comment:
I was doing it because there were a few exit points where a TlsLayer was being created (e.g., for scripts) and I wasn't sure if they might be upgraded with SNI (and subsequently ignored if desired). But of course I can remove the decorator and move the check directly into _next_layer.

If I move the `check_tcp` up will that impact any of the later cases? before it was happening at step 4, after CONNECT etc.

I'll have a patch incoming soon so you can review.